### PR TITLE
fix(pkg): webserver-oneshot when downloading a large file

### PR DESCRIPTION
**Pull request by moyodiallo**
Fixes https://github.com/ocaml/dune/issues/10741

We should merge this PR only after the merge of #10742.

### CODE REVIEW

File: test/blackbox-tests/test-cases/pkg/gh10741.t [hunk 1]
```diff
@@ -0,0 +1,13 @@
+The issue with the oneshot webserver used for testing
+package management, where attempting to send a large file causes the
+connection to be reset before the transfer completes, is now fixed.
+
+Make a large file:
+  $ dd if=/dev/zero of=./large-file iflag=fullblock,count_bytes status=none count=64M
```
Comment (moyodiallo) [inline test/blackbox-tests/test-cases/pkg/gh10741.t:6]: It was tested with count=10000M and it works.

File: test/blackbox-tests/test-cases/pkg/gh10741.t [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+Run the oneshot webserver to serve the large file:
+  $ webserver_oneshot --content-file ./large-file --port-file port.txt &
+  $ until test -f port.txt ; do sleep 0.1; done
```
Comment (maiste) [inline test/blackbox-tests/test-cases/pkg/gh10741.t:10]: I'm not sure to understand: why you need this file?

Comment (moyodiallo) [reply test/blackbox-tests/test-cases/pkg/gh10741.t:10]: Which file ?

Comment (moyodiallo) [reply test/blackbox-tests/test-cases/pkg/gh10741.t:10]: If it is the test file `gh10741.t`, it coming from https://github.com/ocaml/dune/pull/10742 which reproduced the issue.

Comment (Leonidas-from-XIV) [reply test/blackbox-tests/test-cases/pkg/gh10741.t:10]: I think we should merge #10742 first and rebase the fix on it.

Comment (moyodiallo) [reply test/blackbox-tests/test-cases/pkg/gh10741.t:10]: For sure, we could go this.

Comment (maiste) [reply test/blackbox-tests/test-cases/pkg/gh10741.t:10]: Sorry, my question wasn't well written and clear. Why do you need to test the presence of the `port.txt` file in the test?
```
$ until test -f port.txt ; do sleep 0.1; done
```
If it is executed sequentially, we shouldn't have to wait, no?

Comment (Leonidas-from-XIV) [reply test/blackbox-tests/test-cases/pkg/gh10741.t:10]: The server is started in the background (`&`) thus by the time the next line runs, the file might not yet be created, so the next line waits until the server is up and creates the port file (which is its way of signalling "I'm ready to accept a connection").

The server can't be started in the foreground, as then the test never terminates.

**Comment by rgrinberg** [comment]
Fixed in #10753

**Pull request closed without merging**